fix: Remove extra published_state property in the configurator state

ptbrowne opened the pull request:
Made a silly mistake where I put published_state in the config and also
in the configurator state. Published_state should only be in the config,
not the state.

This resulted in old charts not being able to edited/copied/viewed due
to missing property while decoding configurator state.

Fix https://github.com/visualize-admin/visualization-tool/issues/1354

bprusinowski reviewed (changes requested):
@ptbrowne I think you also need to remove the migration then if the property is no longer in the config?

I also wonder why this bug appeared when migration was correctly added? Shouldn't the charts be migrated correctly, maybe we miss a `migrateConfiguratorState` call somewhere?

ptbrowne commented:
Thanks Bartosz, good point, I removed the migration.

> I also wonder why this bug appeared when migration was correctly added? 

Thanks also for wondering this 😅 So I investigated, and I had forgotten to bump the final version in `CONFIGURATOR_STATE_VERSION`. So the last migration had not been applied. With the mixing up of the attribute in configurator state, and config, I thought it was working when it was not. I added a test to make sure we do not forget to update the version.

bprusinowski reviewed (approved):
LGTM, thanks 🎉
